From ad3a9291c341c1239eac8a8e864e2e775f757848 Mon Sep 17 00:00:00 2001
From: Reto Schneider <reto.schneider@husqvarnagroup.com>
Date: Fri, 24 Dec 2021 04:17:53 +0100
Subject: [PATCH] [sg noup] Link against shared libraries

Refreshing the preprocessed nng header is needed because we no longer
build against a predefined version.

Distutils makes it very hard to support customizable builds. Trying to
change the build system to support both, static and dynamic building
turned out to be pretty much impossible.

Therefore, all of this has works-for-me quality, can not be upstreamed.
---
 build_pynng.py | 8 +++-----
 setup.py       | 4 ----
 2 files changed, 3 insertions(+), 9 deletions(-)

diff --git a/build_pynng.py b/build_pynng.py
index 3215129..ca7e0dc 100644
--- a/build_pynng.py
+++ b/build_pynng.py
@@ -22,9 +22,8 @@ if sys.platform == 'win32':
     # system libraries determined to be necessary through trial and error
     libraries = ['Ws2_32', 'Advapi32']
 else:
-    objects = ['./nng/build/libnng.a', "./mbedtls/prefix/lib/libmbedtls.a",
-               "./mbedtls/prefix/lib/libmbedx509.a", "./mbedtls/prefix/lib/libmbedcrypto.a"]
-    libraries = ['pthread']
+    objects = []
+    libraries = ['pthread', 'nng']
     machine = os.uname().machine
     # this is a pretty heuristic... but let's go with it anyway.
     # it would be better to get linker information from cmake somehow.
@@ -37,7 +36,7 @@ ffibuilder.set_source(
     r""" // passed to the real C compiler,
          // contains implementation of things declared in cdef()
          #define NNG_DECL
-         #define NNG_STATIC_LIB
+         #define NNG_SHARED_LIB
          #include <nng/nng.h>
          #include <nng/protocol/bus0/bus.h>
          #include <nng/protocol/pair0/pair.h>
@@ -57,7 +56,6 @@ ffibuilder.set_source(
     libraries=libraries,
     # library_dirs=['nng/build/Debug',],
     # (more arguments like setup.py's Extension class:
-    include_dirs=['nng/include'],
     extra_objects=objects,
 )
 
diff --git a/setup.py b/setup.py
index 20cd02d..f9aaeb0 100644
--- a/setup.py
+++ b/setup.py
@@ -156,10 +156,6 @@ tests_require = [
 ]
 
 setuptools.setup(
-    cmdclass={
-        'build_py': BuildPyCommand,
-        'build_ext': BuildExtCommand,
-    },
     name='pynng',
     version=__version__,
     author='Cody Piersall',
-- 
2.30.2

